src/Simplex_tree/include/gudhi/Simplex_tree/Simplex_tree_star_simplex_iterators.h

```diff
@@ -73,7 +73,12 @@ class Simplex_tree_optimized_cofaces_rooted_subtrees_simplex_iterator
     // coface of simp_
     bool operator()(typename SimplexTree::Hooks_simplex_base& curr_hooks) {
       Node& curr_node = static_cast<Node&>(curr_hooks);
-      auto sh = cpx_->simplex_handle_from_node(curr_node);
+      Simplex_handle sh;
+      if constexpr (SimplexTree::Options::simplex_handle_strong_validity){
+        sh = cpx_->simplex_handle_from_node(curr_node, simp_.front());
+      } else {
+        sh = cpx_->simplex_handle_from_node(curr_node);
+      }
```

Comment:
if we stick to the implementation for map that needs the vertex, I guess we could always pass front as 2nd argument, and simply inside simplex_handle_from_node have two implementations (with `if constexpr[), one of which ignores the second argument. But let's see if the version without vertex works first.